marklogic-data-hub-central/ui/src/components/facet/facet.test.tsx

```diff
@@ -97,4 +99,35 @@ describe("Facet component", () => {
     expect(document.querySelector('[data-testid=sales_region-toggle] i svg')).toHaveStyle('transform: rotate(180deg);');
   });
 
+  it("Search link shown when facet number greater than limit" , () => {
```

Comment:
I would recommend 3 things:
1. Create a default facetProps under assets/mock-data to be used by all the facet tests
2. Move the < 20 test into this one. facetValues is already mocked out and has 12 items. Can use the default props to verify for < 20 condition
3. Do facetValues.push to get it above 20, rerender with updated facetValues and test for > 20 conddition. 

Refer https://wiki.marklogic.com/display/ENGINEERING/Writing+Data+Hub+Tests#WritingDataHubTests-RTLTests for guidance.

Reply:
@bsrikan for tests of this functionality, I'd prefer to be able to set the limit as a constant and then key off of that for the tests (to make it clear what the limit is and make it easy to change). I can rewrite to test facet values for LIMIT-1, rerender, and then test values for LIMIT.

Reply:
You can still test the way you want to. You are going to use the default props which has less than 20 values, which is how `expect(queryByLabelText('popover-search-label')).not.toBeInTheDocument();` is being tested now. Only difference with my recommendation is both conditions will be asserted in same test making the expected behavior more apparent.

```
    const { getByLabelText, rerender } = render(<Facet {...data.facetProps} />)
    // Search link not shown for facets < 20
    expect(queryByLabelText('popover-search-label')).not.toBeInTheDocument();
    
    // Search link shown for facets >= 20
    const LIMIT =  20;
    const f = {
      "name": "facetName",
      "count": 123,
      "value": "facetValue"
    };
    let facetVals: any = [];
    for (let i = 0; i < LIMIT; i++) {
      facetVals.push(f); //run time appending to facetValues in assets/mock-data/entity-table
    }
    rerender(<Facet {...data.facetProps} facetValues={facetVals}/>)
    expect(getByLabelText('popover-search-label')).toBeInTheDocument();
```

Reply:
@bsrikan, thanks! I refactored the tests using a new facetProps default object, much cleaner.